test_parameter_example fails consistently with hypothesis

Detected on github workflows, macOS-latest python 3.12 runner.

```
    | Traceback (most recent call last):
    |   File "/Users/runner/work/ert/ert/tests/integration_tests/test_parameter_example.py", line 377, in test_parameter_example
    |     p.check(io_source, mask, num_realizations)
    |   File "/Users/runner/work/ert/ert/tests/integration_tests/test_parameter_example.py", line 257, in check
    |     np.testing.assert_allclose(
    |   File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/numpy/testing/_private/utils.py", line 1504, in assert_allclose
    |     assert_array_compare(compare, actual, desired, err_msg=str(err_msg),
    |   File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/contextlib.py", line 81, in inner
    |     return func(*args, **kwds)
    |            ^^^^^^^^^^^^^^^^^^^
    |   File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/numpy/testing/_private/utils.py", line 797, in assert_array_compare
    |     raise AssertionError(msg)
    | AssertionError: 
    | Not equal to tolerance rtol=0.0001, atol=5e-05
    | 
    | Mismatched elements: 2 / 2 (100%)
    | Max absolute difference: 1.
    | Max relative difference: 0.5
    |  x: MaskedArray([[[inf, 3.]]], dtype=float32)
    |  y: array([[[2., 2.]]], dtype=float32)
    | Draw 1 (shape): (1, 1, 2)
    | Draw 2 (actnum): [0, 1]
    ...
```


https://github.com/equinor/ert/actions/runs/9223764957/job/25377743795

in test_parameter_example.py:

```
@reproduce_failure('6.102.6', b'AXicY2BgZGBkZIADATZmIMkB4TAyMTGB6N/eCz34ln0Aspo/RjyJ////P0IDIxjCATNCHA6YMESAYgDwtQoX')
@settings(max_examples=10)
@given(
    io_source=st.builds(IoProvider, data=st.data()),
    grid_format=st.sampled_from(["grid", "egrid"]),
    summary=summaries(
        start_date=st.just(datetime(1996, 1, 1)),
        time_deltas=st.just([1.0, 2.0]),
        summary_keys=st.just(["FOPR"]),
    ),
    num_realizations=st.integers(min_value=1, max_value=10),
    parameters=st.lists(
        st.one_of(field_parameters(), st.builds(SurfaceParameter, names)),
        unique_by=lambda x: x.name,
        max_size=3,
    ),
)
def test_parameter_example(
    io_source, grid_format, summary, tmp_path_factory, num_realizations, parameters
):
    tmp_path = tmp_path_factory.mktemp("parameter_example")
    note(f"Running in directory {tmp_path}")
...
```


⚠️ Hypothesis did report two issues, but the other one I could not verify failing.
```
@reproduce_failure('6.102.6', b'AXicY2BgZGBkZIADATZmIMkB4TAyMTGB6N/eCz34ln0Aspo/RjyJ////P0IDIxhiAiQxJgwRoBgA8DoKFA==')
```

Is this mac specific? I get the following when trying to reproduce:
```
E   hypothesis.errors.DidNotReproduce: Expected the test to raise an error, but it completed successfully.
```

--- a/tests/integration_tests/test_parameter_example.py
+++ b/tests/integration_tests/test_parameter_example.py
@@ -264,7 +264,7 @@ class FieldParameter:
 
 @st.composite
 def field_parameters(draw):
-    min = draw(
+    min_value = draw(
         st.one_of(
             st.none(),
             st.floats(
@@ -272,11 +272,18 @@ def field_parameters(draw):
             ),
         )
     )
-    max = st.one_of(
+    max_value = st.one_of(
         st.none(),
-        st.floats(min_value=2.0, max_value=1e9, allow_nan=False, allow_infinity=False),
+        st.floats(
+            min_value=max([2.0, min_value if min_value is not None else 0.0]),
+            max_value=1e9,
+            allow_nan=False,
+            allow_infinity=False,
+        ),
     )
-    return draw(st.builds(FieldParameter, name=names, min=st.just(min), max=max))
+    return draw(
+        st.builds(FieldParameter, name=names, min=st.just(min_value), max=max_value)
+    )
 
 
 @dataclass

